neo/Ledger/MemoryPool.cs

```diff
@@ -286,31 +287,40 @@ internal bool TryAdd(UInt256 hash, Transaction tx)
 
             if (_unsortedTransactions.ContainsKey(hash)) return false;
 
+            List<Transaction> removedTransactions = new List<Transaction>();
             _txRwLock.EnterWriteLock();
             try
             {
                 _unsortedTransactions.Add(hash, poolItem);
 
                 SortedSet<PoolItem> pool = tx.IsLowPriority ? _sortedLowPrioTransactions : _sortedHighPrioTransactions;
                 pool.Add(poolItem);
-                RemoveOverCapacity();
+                RemoveOverCapacity(removedTransactions);
             }
             finally
             {
                 _txRwLock.ExitWriteLock();
             }
 
+            foreach (IMemoryPoolTxObserverPlugin plugin in Plugin.TxObserverPlugins)
+            {
+                plugin.AddedTransaction(poolItem.Tx);
+                if (removedTransactions.Count > 0)
+                    plugin.RemovedTransactions(MemoryPoolTxRemovalReason.CapacityExceeded, removedTransactions);
+            }
+
             return _unsortedTransactions.ContainsKey(hash);
         }
 
-        private void RemoveOverCapacity()
+        private void RemoveOverCapacity(List<Transaction> removedTransactions)
```

Comment:
Change it to return the removed transactions?

Reply:
I think that is a reasonable suggestion. I'll make that change.